calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/metadata/DocumentDbTableSchemaGeneratorTest.java

```diff
@@ -617,7 +617,7 @@ void testComplexSingleLevelArray() {
     void testComplexTwoLevelArray() {
         final Map<String, String> tableNameMap = new HashMap<>();
         final BsonDocument document = BsonDocument.parse(
-                "{ \"_id\" : \"key\", \"array\" : [ [1, 2, 3 ], [ 4, 5, 6 ] ]}");
+                "{ \"_id\" : \"key\", \"array\" : [ [1, 2, 3 ], [ 4, 5, 6 ], [7, 8, 9 ] ]}");
```

Comment:
Added additional nested array entry here. Before changing if statement to use table name and moving adding foreign key to outside if statement, the second array index was being added to foreign key list multiple times. With testing with this extra entry, no duplicate columns were produced BUT the primary key index was showing 4 instead of 3.